flixel/input/actions/FlxSteamController.hx

```diff
@@ -0,0 +1,342 @@
+					break;
+				}
+			}
+		}
+		else
+		{
+			Steam.controllers.activateActionSet(SteamControllerHandle, ActionSetHandle);
+		}
+		#end
+	}
+	
+	private static function getFirstActiveHandle():Int
+	{
+		#if (cpp && steamwrap && haxe_ver > "3.2")
+		if (controllers == null) return -1;
+		for (i in 0...controllers.length)
+		{
+			if (controllers[i].active)
+			{
+				return controllers[i].handle;
+			}
+		}
+		#end
+		return -1;
+	}
+	
+	private static function getConnectedControllers():Array<Int>
+	{
+		#if (cpp && steamwrap && haxe_ver > "3.2")
+		if (Steam.controllers == null) return [];
+		var arr = Steam.controllers.getConnectedControllers();
+		
+		for (i in 0...Steam.controllers.MAX_CONTROLLERS)
+		{
+			if (i < arr.length && arr[i] >= 0)
+			{
+				var index = arr[i];
+				controllers[index].handle = index;
+			}
+			
+			controllers[i].connected.update();
+			if (arr.indexOf(controllers[i].handle) != -1)
+			{
+				controllers[i].connected.press();
+			}
+			else
+			{
+				controllers[i].connected.release();
+			}
+		}
+		
+		return arr;
+		#else
+		return [];
+		#end
+	}
+	
+	#if (cpp && steamwrap && haxe_ver > "3.2")
+	private static function getAnalogActionData(controller:Int, action:Int, ?data:ControllerAnalogActionData):ControllerAnalogActionData
+	{
+		if (Steam.controllers == null) return data;
+		data = Steam.controllers.getAnalogActionData(controller, action, data);
+		if (controller >= 0 && controller < controllers.length)
+		{
+			if (data.bActive > 0 && data.x != 0 || data.y != 0)
+			{
+				controllers[controller].active = true;
+			}
+		}
+		return data;
+	}
+	#else
+	private static function getAnalogActionData(controller:Int, action:Int, ?data:Dynamic):Dynamic
+	{
+		return null;
+	}
+	#end
+	
+	#if (cpp && steamwrap && haxe_ver > "3.2")
+	private static function getDigitalActionData(controller:Int, action:Int):ControllerDigitalActionData 
+	{
+		if (Steam.controllers == null) return 0;
+		var data = Steam.controllers.getDigitalActionData(controller, action);
+		if (controller >= 0 && controller < controllers.length)
+		{
+			if (data.bActive && data.bState)
+			{
+				controllers[controller].active = true;
+			}
+		}
+		return data;
+	}
+	#else
+	private static function getDigitalActionData(controller:Int, action:Int):DigitalActionData
+	{
+		return
+		{
+			bActive:false,
+			bState:false
+		};
+	}
+	#end
+	
+	private static inline function getAnalogActionHandle(name:String):Int
+	{
+		#if (cpp && steamwrap && haxe_ver > "3.2")
+		if (Steam.controllers == null) return -1;
+		return Steam.controllers.getAnalogActionHandle(name);
+		#else
+		return -1;
+		#end
+	}
+	
+	private static inline function getDigitalActionHandle(name:String):Int
+	{
+		#if (cpp && steamwrap && haxe_ver > "3.2")
+		if (Steam.controllers == null) return -1;
+		return Steam.controllers.getDigitalActionHandle(name);
+		#else
+		return -1;
+		#end
+	}
+}
+
+typedef DigitalActionData =
```

Comment:
Would prefer if this was a class. Removes any doubt whether it will result in fast native code.